src/main/java/com/autotune/analyzer/utils/GsonUTCDateAdapter.java

```diff
@@ -0,0 +1,53 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, IBM Corporation and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.analyzer.utils;
+
+import com.google.gson.*;
+
+import java.lang.reflect.Type;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+/**
+ * UTC data handler for GSON
+ */
+public class GsonUTCDateAdapter implements JsonSerializer<Date>, JsonDeserializer<Date> {
+
+    private final DateFormat dateFormat;
+
+    public GsonUTCDateAdapter() {
+        dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US);      //This is the format I need
```

Comment:
Why is the locale here the `US`? Also what happens when we send in dates not in the UTC TZ

Reply:
@dinogun thanks ofr pointing it out. as per syntax we have to specify some locale and  I just changed it to root , The root locale is the locale whose language, country, and variant are empty ("") strings.

Input Json should follow this format if not throw error
if Z is not specifed  then API will throw error 
But it allows if we specify it as +00:00  
in similar fashion input request are allowed to specify UTC as

"−05:00" for New York on standard time ([UTC−05:00](https://en.wikipedia.org/wiki/UTC%E2%88%9205:00))
    "−04:00" for New York on daylight saving time ([UTC−04:00](https://en.wikipedia.org/wiki/UTC%E2%88%9204:00))
    "+00:00" (but not "−00:00") for London on standard time ([UTC±00:00](https://en.wikipedia.org/wiki/UTC%C2%B100:00))
    "+02:00" for Cairo ([UTC+02:00](https://en.wikipedia.org/wiki/UTC%2B02:00))
    "+05:30" for Mumbai ([UTC+05:30](https://en.wikipedia.org/wiki/UTC%2B05:30))
    "+14:00" for Kiribati ([UTC+14:00](https://en.wikipedia.org/wiki/UTC%2B14:00))

I hope this answers.

Reply:
I am thinking that we need a short para in the doc MD on the timestamp format, so that the users know exactly what is expected. Please add that as part of this PR